pinot-integration-tests/src/test/java/org/apache/pinot/compat/tests/SqlResultComparator.java

```diff
@@ -113,9 +126,147 @@ public static boolean areEqual(JsonNode actual, JsonNode expected, String query)
+      String expectedOrderByColumnValues = "";
+      String actualOtherColumnValues = "";
+      String expectOtherColumnValues = "";
+      ArrayNode actualValue = (ArrayNode) actualElements.get(i).get(FIELD_VALUE);
+      ArrayNode expectedValue = (ArrayNode) expectedElements.get(i).get(FIELD_VALUE);
+
+      for (int j = 0; j < actualValue.size(); j++) {
+        if (orderByColumnIndexs.contains(j)) {
+          actualOrderByColumnValues += ", " + actualValue.get(j).toString();
+          expectedOrderByColumnValues += ", " + expectedValue.get(j).toString();
+        } else {
+          actualOtherColumnValues += ", " + actualValue.get(j).toString();
+          expectOtherColumnValues += ", " + expectedValue.get(j).toString();
+        }
+      }
+      lastGroupOrderByColumnValues = actualOrderByColumnValues;
+
+      actualOrderByColumnValuesToOtherColumnValuesMap.
+          computeIfAbsent(actualOrderByColumnValues, k -> new LinkedList<>()).
+          add(actualOtherColumnValues);
+      expectedOrderByColumnValuesToOtherColumnValuesMap.
+          computeIfAbsent(expectedOrderByColumnValues, k -> new LinkedList<>()).
+          add(expectOtherColumnValues);
+    }
+
+    if (!actualOrderByColumnValuesToOtherColumnValuesMap.keySet().
+        equals(expectedOrderByColumnValuesToOtherColumnValuesMap.keySet())) {
+      LOGGER.error("The results of the ordered query has different groups, actual: {}, expected: {}",
+          actualOrderByColumnValuesToOtherColumnValuesMap.keySet(),
+          expectedOrderByColumnValuesToOtherColumnValuesMap.keySet());
+      return false;
+    }
+
+    for (Map.Entry<String, List<String>> entry : actualOrderByColumnValuesToOtherColumnValuesMap.entrySet()) {
+      String orderByColumnValues = entry.getKey();
+      // For the last group, skip the value comparison for other columns.
+      if (orderByColumnValues.equals(lastGroupOrderByColumnValues)) {
+        continue;
+      }
+      List<String> actualOtherColumnValues = entry.getValue();
+      List<String> expectedOtherColumnValues =
+          expectedOrderByColumnValuesToOtherColumnValuesMap.get(orderByColumnValues);
+      Collections.sort(actualOtherColumnValues);
+      Collections.sort(expectedOtherColumnValues);
+      if (!actualOtherColumnValues.equals(expectedOtherColumnValues)) {
+        LOGGER.error(
+            "The results of the ordered query has different non-order-by column values for group: {}, actual: {}, expected: {}",
+            orderByColumnValues, actualOtherColumnValues, expectedOtherColumnValues);
+        return false;
+      }
+    }
+
+    return true;
+  }
+
+  private static boolean areNonOrderByQueryElementsEqual(List<String> actualElementsSerialized,
+      List<String> expectedElementsSerialized) {
+    // sort elements
+    actualElementsSerialized.sort(null);
+    expectedElementsSerialized.sort(null);
+    if (!actualElementsSerialized.equals(expectedElementsSerialized)) {
```

Comment:
Pinot doesn't guarantee deterministic results (rows) for multiple executions of queries without ORDER BY. How do we know that a query without ORDER BY will return same results when run twice ?

Reply:
For non-orderby queries, the result file should specify the isSuperset flag. If that is done, we will never hit this case. 
@mqliang  it may be a good idea to add that as a comment here, so that if people see failures, they can update the result file.

Reply:
done